presto-main/src/main/java/com/facebook/presto/operator/PipelineContext.java

```diff
@@ -343,6 +343,11 @@ public CounterStat getOutputPositions()
         return stat;
     }
 
+    public PipelineStatus getPipelineStatus()
+    {
+        return new PipelineStatus(ImmutableList.copyOf(this.drivers));
```

Comment:
Make the `PipelineStatus` constructor as dumb as possible. Put logic here instead.